class_name UpgradePanel
extends PanelContainer

signal mouse_clicked

@export var shiny_shader = ShaderMaterial

@onready var title: Label = $VBoxContainer/MarginContainer3/Label
@onready var icon: TextureRect = $VBoxContainer/MarginContainer/TextureRect
@onready var text: RichTextLabel = $VBoxContainer/MarginContainer2/RichTextLabel

var _color_table: ColorTable = preload("res://Systems/color_table.tres")


func _gui_input(event: InputEvent) -> void:
	if event.is_action_pressed("left_click"):
		mouse_clicked.emit()


func _input(event):
	if event.is_action_pressed("ui_accept") and has_focus():
		mouse_clicked.emit()


func make_shiny():
	material = shiny_shader


func recolor(color_name: String):
	if not color_name in _color_table.colors:
		return

	material = material.duplicate()

	var colors: Array = _color_table.colors[color_name]
	for i in len(colors):
		material.set_shader_parameter("replace_%s" % i, colors[i])
